app/models/product_consume_log.rb

```diff
@@ -56,7 +56,7 @@ def record_monthly_amount(consume_date)
     end
   end
 
-  def calculate_average_period
+  def average_period
     consume_logs = ProductConsumeLog.where('product_id=?', product_id).where.not(use_ended_at: nil)
```

Comment:
```suggestion
    consume_logs = ProductConsumeLog.where(product_id: product_id).where.not(use_ended_at: nil)
```

joinをしてなければ上記のような書き方ができると思います。

また、同じ行が他にもあるのでメソッドやscopeで共通化するといいと思います。